### PR TITLE
NAP-210 service hours fix

### PR DESCRIPTION
**Pull request by rckopteri**

### CODE REVIEW

File: ote/src/cljc/ote/time.cljc [hunk 1]
```diff
@@ -24,6 +25,24 @@
          cljs-time/to-default-time-zone
          (format/unparse (format/formatter "dd.MM.yyyy HH:mm"))))))
 
+#?(:cljs
+   (defn js-time-to-string [time]
+     (if  (nil? time)
+       " " ;: if nil - print empty string
```
Comment (tatut) [inline ote/src/cljc/ote/time.cljc]: not an empty string, contains a single space.

File: ote/src/cljc/ote/time.cljc [hunk 2]
```diff
@@ -24,6 +25,24 @@
          cljs-time/to-default-time-zone
          (format/unparse (format/formatter "dd.MM.yyyy HH:mm"))))))
 
+#?(:cljs
+   (defn js-time-to-string [time]
```
Comment (tatut) [inline ote/src/cljc/ote/time.cljc]: Isn't this formatting? Should it be called `format-js-time` or something like that. Or have a function `from-js-time` that turns JS Date object to OTE Time record and call format on that.

File: ote/src/cljc/ote/time.cljc [hunk 3]
```diff
@@ -24,6 +25,24 @@
...
+     (if  (nil? time)
+       " " ;: if nil - print empty string
+       (->> time
+            cljs-time/to-default-time-zone
+            (format/unparse (format/formatter "HH:mm:ss"))))))
+
+#?(:cljs
+   (defn js-time-from-db-time [db-time]
```
Comment (tatut) [inline ote/src/cljc/ote/time.cljc]: Perhaps `to-js-time` or something like that.

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 4]
```diff
@@ -194,6 +194,18 @@
...
+(defmethod field :time-picker [{:keys [update! ok-label cancel-label default-time] :as opts} data]
+  (let [time-picker-time (if (= nil? data) default-time data)]
+  [ui/time-picker
+   {:format "24hr"
+    :cancel-label cancel-label
+    :ok-label ok-label
+    :minutes-step 5
+    :default-time (time/js-time-from-db-time time-picker-time)
+    :on-change (fn [event value]
+                 (update! (with-meta (time/parse-time (time/js-time-to-string value))
```
Comment (tatut) [inline ote/src/cljs/ote/ui/form_fields.cljs]: Does a time-picker need an incomplete value? User can't type in this component.

File: ote/src/cljc/ote/time.cljc (hunk 2 above)
Comment (rckopteri) [reply ote/src/cljc/ote/time.cljc]: We already have time/parse-time that converts string to ->Time object.
So I only change function name to format-js-time.

File: ote/src/cljc/ote/time.cljc (hunk 1 above)
Comment (rckopteri) [reply ote/src/cljc/ote/time.cljc]: Nice catch. Thanks.

**Pull request merged**